equinity-historical-data/src/main/java/uk/gov/justice/laa/crime/equinity/historicaldata/mapper/Crm4Mapper.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.justice.laa.crime.equinity.historicaldata.mapper;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import uk.gov.justice.laa.crime.equinity.historicaldata.generated.dto.Crm4DetailsDTO;
+import uk.gov.justice.laa.crime.equinity.historicaldata.model.Crm4DetailsModel;
+
+@Mapper(componentModel = "spring")
+public interface Crm4Mapper {
+
+    @Mapping(target="greaterValue", source="applyingge100")
+    @Mapping(target="postMortemExamination", source="post_mortem_app")
+    @Mapping(target="caseDetails.isPOCA", source="is_case_subject_to_poca")
+    @Mapping(target="caseDetails.firm.urn", source="urn")
+    @Mapping(target="caseDetails.firm.firmName", source="firm_name")
+    @Mapping(target="caseDetails.firm.firmAddress", source="firm_address")
+    @Mapping(target="caseDetails.firm.firmPhone", source="firm_phone")
+    @Mapping(target="caseDetails.firm.firmSupplierName", source="provider_account")
+    @Mapping(target="caseDetails.firm.firmSupplierNo", source="firm_supplier_no")
+    @Mapping(target="caseDetails.firm.firmContactName", source="contact_name")
+    @Mapping(target="caseDetails.firm.firmSolicitorName", source="solicitorname")
+
+    @Mapping(target="caseDetails.solicitorDetails.solicitorName", source="solicitorname")
+    @Mapping(target="caseDetails.solicitorDetails.solicitorReference", source="solicitorid")
+
+    @Mapping(target="caseDetails.clientDetails.ufn", source="client_ufn")
+    @Mapping(target="caseDetails.clientDetails.maatNumber", source="maat_number")
+    @Mapping(target="caseDetails.clientDetails.firstName", source="client_forename")
+    @Mapping(target="caseDetails.clientDetails.middleName", expression = "java(null)")
```

Comment:
Just as a remainder, for both of us. On those fields we are currently adding `null`, we need to get confirmation from BAs if they are still required, should be avoided or if they should be added to the Model even those the field does not exists in our test XML

Reply:
I'll remove middlename it is not returned as part of existing data.